### ISSUE TITLE
Add a SettingKey to define the initial project on (re)load

### ISSUE THREAD

**Issue by jaceklaskowski** [sbt/sbt#1224]
From https://groups.google.com/d/msg/sbt-dev/dB1TmFN8pvw/4JF_3CzPAJYJ:

_"In either case, feel free to open an enhancement ticket about this if it's causing signfiicant issues.  We can probably create a setting the `reload` command will read when setting the initial project, rather than always using "root"."_

**Comment by jaceklaskowski** [comment]
Submitted https://github.com/sbt/sbt/pull/1305 to get this going. Please review and guide me to make it complete. I call myself stuck on trying to understand the sbt sources and the way projects are loaded.

**Comment by dwijnand** [comment]
Copy (and tweaked) workaround from #2405:

```scala
onLoad in Global ~= (_ andThen ("project foo" :: _))
```

**Comment by jaceklaskowski** [comment]
It could've been easier, but the workaround by @dwijnand makes a lot of sense to me and so I'm closing  the issue.

**Comment by dwijnand** [comment]
I agree it could and should be easier. Do you mind if we use this issue to track the addition of a more formal way to define the initial project?
